cmd/system-probe/config/config.go

```diff
@@ -99,6 +99,20 @@ func newSysprobeConfig(configPath string, loadSecrets bool) (*Config, error) {
 	// load the configuration
 	_, err := aconfig.LoadCustom(aconfig.SystemProbe, "system-probe", loadSecrets, aconfig.Datadog.GetEnvVars())
 	if err != nil {
+		if errors.Is(err, os.ErrPermission) {
+			if runtime.GOOS == "darwin" {
+				log.Infof("Error loading config: %v (check config file permissions for dd-agent user)", err)
+			} else {
+				log.Warnf("Error loading config: %v", err)
+			}
+		} else {
```

Comment:
Your top-level if/else is mixed up. The message about permissions should only be in the `if` block.

Reply:
Oh thank you! Completely missed that 😄 Just pushed a fix